examples/azure/client_factory.md

```diff
@@ -0,0 +1,133 @@
+## Azure SDK Client Factory 
+
+This documentation provides and overview of the `client_factory.go` module, targeted use cases, and behaviors. 
+
+## Multi-cloud environment support
```

Comment:
These are solid docs, but does putting it in the `examples` folder make sense? Or should this be top-level documentation part of the [Terratest documentation site](https://terratest.gruntwork.io/docs/)? Note that the site is generated from Markdown repos in the same repo, so this would be mostly about moving these docs. E.g., There could be an "Azure authentication" page under "Testing best practices."

Reply:
Agreed on docs...we picked an initial spot that didn't get us into messing with the docs jekyl site at the same time. @rguthriemsft - thoughts here?

Reply:
Discussion of docs is happening [in Slack](https://gruntwork-io.slack.com/archives/C015S5V8DE2/p1602110186026800?thread_ts=1601681297.020100&cid=C015S5V8DE2), so I'll mark this one as unresolved for now.